### ISSUE TITLE
Error with npm install and node 0.12

### ISSUE THREAD

**Issue by rcollette** [DefinitelyTyped/tsd#124]
```
npm install tsd --save-dev
npm WARN engine tsd@0.5.7: wanted: {"node":">= 0.10.0 < 0.11.0"} (current: {"nod
e":"0.12.0","npm":"2.5.1"})
npm WARN `git config --get remote.origin.url` returned wrong result (git://githu
b.com/Bartvds/update-notifier/)
npm ERR! git clone git://github.com/Bartvds/update-notifier/ Cloning into bare r
epository 'C:\Users\n1200632d\AppData\Roaming\npm-cache\_git-remotes\git-github-
com-Bartvds-update-notifier--a420e038'...
npm ERR! git clone git://github.com/Bartvds/update-notifier/ fatal: unable to co
nnect to github.com:
npm ERR! git clone git://github.com/Bartvds/update-notifier/ github.com[0: 192.3
0.252.130]: errno=No error
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nodejs
\\node_modules\\npm\\bin\\npm-cli.js" "install" "tsd" "--save-dev"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1
npm ERR! code 128

npm ERR! Command failed: git -c core.longpaths=true clone --template=C:\Users\n1
200632d\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror git://github.
com/Bartvds/update-notifier/ C:\Users\n1200632d\AppData\Roaming\npm-cache\_git-r
emotes\git-github-com-Bartvds-update-notifier--a420e038
npm ERR! Cloning into bare repository 'C:\Users\n1200632d\AppData\Roaming\npm-ca
che\_git-remotes\git-github-com-Bartvds-update-notifier--a420e038'...
npm ERR! fatal: unable to connect to github.com:
npm ERR! github.com[0: 192.30.252.130]: errno=No error
npm ERR!
npm ERR!
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <http://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Code\Internet Team Projects\vs2012\MyHomeWorks\MyHomeWorks - Enh
ancement Branch\MyHomeWorks.Web\npm-debug.log

```

**Comment by kentcooper** [comment]
I had to change my package.json to use the "next" version of both tsd and grunt-tsd after I upgraded to 12.0. Seems to have solved most of my issues

**Comment by rcollette** [comment]
I've been able to reinstall tsd now.  However, I did get a whole lot of warning during the build of buffer tools???

```

\
C:\Users\rich\AppData\Roaming\npm\node_modules\tsd\node_modules\bufferstream\nod
e_modules\buffertools>node "C:\Users\rich\AppData\Roaming\npm\node_modules\npm\b
in\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild
child_process: customFds option is deprecated, use stdio instead.
Building the projects in this solution one at a time. To enable parallel build,
please add the "/m" switch.
  buffertools.cc
c:\users\rich\appdata\roaming\npm\node_modules\tsd\node_modules\bufferstream\no
de_modules\buffertools\BoyerMoore.h(35): warning C4267: '=' : conversion from '
size_t' to 'int', possible loss of data [C:\Users\rich\AppData\Roaming\npm\node
_modules\tsd\node_modules\bufferstream\node_modules\buffertools\build\buffertoo
ls.vcxproj]
c:\users\rich\appdata\roaming\npm\node_modules\tsd\node_modules\bufferstream\no
de_modules\buffertools\BoyerMoore.h(57): warning C4267: '=' : conversion from '
size_t' to 'int', possible loss of data [C:\Users\rich\AppData\Roaming\npm\node
_modules\tsd\node_modules\bufferstream\node_modules\buffertools\build\buffertoo
ls.vcxproj]
c:\users\rich\appdata\roaming\npm\node_modules\tsd\node_modules\bufferstream\no
de_modules\buffertools\BoyerMoore.h(61): warning C4267: 'initializing' : conver
sion from 'size_t' to 'const int', possible loss of data [C:\Users\rich\AppData
\Roaming\npm\node_modules\tsd\node_modules\bufferstream\node_modules\buffertool
s\build\buffertools.vcxproj]
c:\users\rich\appdata\roaming\npm\node_modules\tsd\node_modules\bufferstream\no
de_modules\buffertools\BoyerMoore.h(62): warning C4267: 'initializing' : conver
sion from 'size_t' to 'const int', possible loss of data [C:\Users\rich\AppData
\Roaming\npm\node_modules\tsd\node_modules\bufferstream\node_modules\buffertool
s\build\buffertools.vcxproj]
c:\users\rich\appdata\roaming\npm\node_modules\tsd\node_modules\bufferstream\no
de_modules\buffertools\BoyerMoore.h(110): warning C4267: '=' : conversion from
'size_t' to 'int', possible loss of data [C:\Users\rich\AppData\Roaming\npm\nod
e_modules\tsd\node_modules\bufferstream\node_modules\buffertools\build\bufferto
ols.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\xlocale(337): wa
rning C4530: C++ exception handler used, but unwind semantics are not enabled.
Specify /EHsc [C:\Users\rich\AppData\Roaming\npm\node_modules\tsd\node_modules\
bufferstream\node_modules\buffertools\build\buffertools.vcxproj]
..\buffertools.cc(214): warning C4267: 'initializing' : conversion from 'size_t
' to 'const int', possible loss of data [C:\Users\rich\AppData\Roaming\npm\node
_modules\tsd\node_modules\bufferstream\node_modules\buffertools\build\buffertoo
ls.vcxproj]
..\buffertools.cc(215): warning C4267: 'initializing' : conversion from 'size_t
' to 'const int', possible loss of data [C:\Users\rich\AppData\Roaming\npm\node
_modules\tsd\node_modules\bufferstream\node_modules\buffertools\build\buffertoo
ls.vcxproj]
..\buffertools.cc(327): warning C4267: '=' : conversion from 'size_t' to 'int32
_t', possible loss of data [C:\Users\rich\AppData\Roaming\npm\node_modules\tsd\
node_modules\bufferstream\node_modules\buffertools\build\buffertools.vcxproj]
..\buffertools.cc(329): warning C4267: '=' : conversion from 'size_t' to 'int32
_t', possible loss of data [C:\Users\rich\AppData\Roaming\npm\node_modules\tsd\
node_modules\bufferstream\node_modules\buffertools\build\buffertools.vcxproj]
..\buffertools.cc(335): warning C4244: 'argument' : conversion from 'const ptrd
iff_t' to 'int32_t', possible loss of data [C:\Users\rich\AppData\Roaming\npm\n
ode_modules\tsd\node_modules\bufferstream\node_modules\buffertools\build\buffer
tools.vcxproj]
..\buffertools.cc(414): warning C4267: 'argument' : conversion from 'size_t' to
 'int', possible loss of data [C:\Users\rich\AppData\Roaming\npm\node_modules\t
sd\node_modules\bufferstream\node_modules\buffertools\build\buffertools.vcxproj
]
     Creating library C:\Users\rich\AppData\Roaming\npm\node_modules\tsd\node_m
  odules\bufferstream\node_modules\buffertools\build\Release\buffertools.lib an
  d object C:\Users\rich\AppData\Roaming\npm\node_modules\tsd\node_modules\buff
  erstream\node_modules\buffertools\build\Release\buffertools.exp
  Generating code
  Finished generating code
```

**Comment by kentcooper** [comment]
No idea on that one. Mine installs without complaint once I got changed the version to next. Were those errors just from installing from the command line like `npm install tsd@next`? I was still getting some errors when I first tried to update because grunt-tsd was still referencing the old version.

**Comment by rcollette** [comment]
I didn't use `@next`.   What I tried previously which failed but which completed this time was `npm uninstall tsd --save-dev` followed by `npm install tsd --save-dev`.   I actually tried that several times with the same error shown initially when I opened the issue.

When I first installed tsd under the previous version of node, it complained that Python was missing.   I installed Python and it compiled that module that is now generation warnings.

I'm going to close this since the install issue is now resolved, but it is still a little concerning to see so many warnings when installing a module.
